public/app/features/apiserver/client.ts

```diff
@@ -34,6 +38,43 @@ export class ScopedResourceClient<T = object, S = object, K = string> implements
     return getBackendSrv().get<Resource<T, S, K>>(`${this.url}/${name}`);
   }
 
+  public watch(opts?: WatchOptions): Observable<ResourceEvent<T, S, K>> {
+    const finalOpts = opts || {};
+    finalOpts.labelSelector = this.parseListOptionsSelector(finalOpts?.labelSelector);
+    finalOpts.fieldSelector = this.parseListOptionsSelector(finalOpts?.fieldSelector);
+    return getBackendSrv()
+      .chunked({
+        url: finalOpts.name ? `${this.url}/${finalOpts.name}` : this.url,
+        params: finalOpts,
+      })
+      .pipe(
+        switchMap((response) => {
+          const events: Array<ResourceEvent<T, S, K>> = [];
+          if (response.ok && response.data) {
+            const decoder = new TextDecoder();
```

Comment:
This will create a new decoder for each chunk, which is not ideal. We can hoist it up and use the same decoder for each chunk.